src/librustc_mir/interpret/place.rs

```diff
@@ -946,8 +946,10 @@ where
             return self.copy_op(src, dest);
         }
         // We still require the sizes to match.
-        assert!(src.layout.size == dest.layout.size,
-            "Size mismatch when transmuting!\nsrc: {:#?}\ndest: {:#?}", src, dest);
+        if src.layout.size != dest.layout.size {
+            error!("Size mismatch when transmuting!\nsrc: {:#?}\ndest: {:#?}", src, dest);
+            throw_unsup!(TransmuteSizeDiff(src.layout.ty, dest.layout.ty));
```

Comment:
Please add a comment explaining why this is even reachable. Also, I don't think there is precedence for using `error!` like this in Miri?

Reply:
This is not supposed to exist. It's either a limitation of how we do type checking... or a fix idea I have will work, but as said before, I'd rather not do that in this PR. I'll leave comments

Reply:
If we can't fix that in this PR then at least there should be a FIXME in this code (maybe even a `delay_span_bug`?), and an issue tracking this.